xray/audit/jas/sastscanner.go

```diff
@@ -0,0 +1,52 @@
+package jas
+
+import (
+	"github.com/jfrog/jfrog-cli-core/v2/xray/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+const (
+	sastScanCommand = "zd"
+	sastScannerType = "analyze-codebase"
+)
+
+type SastScanManager struct {
+	sastScannerResults []utils.SourceCodeScanResult
+	scanner            *AdvancedSecurityScanner
+}
+
+func newSastScanManager(scanner *AdvancedSecurityScanner) (manager *SastScanManager) {
+	return &SastScanManager{
+		sastScannerResults: []utils.SourceCodeScanResult{},
+		scanner:            scanner,
+	}
+}
+
+func (zd *SastScanManager) Run(wd string) (err error) {
+	scanner := zd.scanner
+	if err = zd.runAnalyzerManager(wd); err != nil {
+		return
+	}
+	var workingDirResults []utils.SourceCodeScanResult
+	workingDirResults, err = getSourceCodeScanResults(scanner.resultsFileName, wd, utils.Sast)
```

Comment:
Missing err check

Reply:
We don't need to check err here, this is our code, already checked when we did sarif.Open inside this method, other than that we just parse the Sarif result

Reply:
i believe @omerzi  meant if err != nil check